lib/msf/core/framework.rb

```diff
@@ -199,7 +199,7 @@ def version
   # @return [Metasploit::Framework::DataService::DataProxy]
   def db
     synchronize {
-      @db ||= Metasploit::Framework::DataService::DataProxy.instance
+      @db ||= get_db
```

Comment:
All this is part of moving initialization to one place while removing the singleton logic and cyclic reference (fw->db, db->fw)